code/modules/mob/dead/observer/observer.dm

```diff
@@ -303,6 +303,7 @@ Works together with spawning an observer, noted above.
 	ghost.client?.init_verbs()
 	if(!can_reenter_corpse)// Disassociates observer mind from the body mind
 		ghost.mind = null
+	ghost.client?.player_details.time_of_death = ghost.mind?.current ? mind.current.timeofdeath : world.time
```

Comment:
Couldn't this be expected to always be null in certain scenarios, given that we null the ghost mind out on line 305? I'd much rather we do this _before_ we do the irreversible uncoupling from observer mind and body mind as that seems like the "final" step to me, so just move it above line 304 so we don't always have to keep defaulting to `world.time` for a case that we set ourselves and so that we can actually read the proper time of death instead of constantly falling into that pit

Reply:
It is done that way to prevent abusing `"Ghost"` verb to a respawn, since the mob's `timeofeath` probably would be 0, so, instant respawn.

Should I check `can_reenter_corpse` and override `time_of_death` with uncoupling?

Reply:
sorry, i'm referring specifically to the fact that we use the funny `?` operator in line 306, which is meant to be a "safe failure" in the instance that a typed variable (like `client`) is `null`. we null out the mind explicitly on line 305, so it just seems silly that we null the mind, then try and do some stuff with it. it makes more sense in my mind to put what is currently line 306 above line 304.

hopefully this is a bit more clear. apologies.

Reply:
So, if I do that on line 304, I will need to overwrite `time_of_death` in 306 anyway when decoupling mind from ghost, if `can_reenter_corpse` is false.

Since if I only just put 306 on 304 without any changes, then pressing `"Ghost"` will basically give an ability to respawn.

Reply:
sorry about the late response @larentoun 

i still think that there could be a better way to do it than having to `?` check a variable that we explicitly know can be null in some scenarios but it seems that `time_of_death` is more critical than it might appear.